mediator/esmFlds.F90

```diff
@@ -593,23 +594,47 @@ end subroutine med_fldList_GetFldInfo_general
 
   !================================================================================
 
-  subroutine med_fldList_GetFldInfo_stdname(fldList, fldindex, stdname)
+  subroutine med_fldList_GetFldInfo_stdname(fldList, fldindex_in, stdname_out)
     ! ----------------------------------------------
     ! Get field info
     ! ----------------------------------------------
     type(med_fldList_type) , intent(in)  :: fldList
-    integer                      , intent(in)  :: fldindex
-    character(len=*)             , intent(out) :: stdname
+    integer                      , intent(in)  :: fldindex_in
+    character(len=*)             , intent(out) :: stdname_out
 
     ! local variables
     character(len=*), parameter :: subname='(med_fldList_GetFldInfo_stdname)'
     ! ----------------------------------------------
 
-    stdname   = fldList%flds(fldindex)%stdname
+    stdname_out   = fldList%flds(fldindex_in)%stdname
   end subroutine med_fldList_GetFldInfo_stdname
 
   !================================================================================
 
+  subroutine med_fldList_GetFldInfo_index(fldList, stdname_in, fldindex_out)
+    ! ----------------------------------------------
+    ! Get field info
+    ! ----------------------------------------------
+    type(med_fldList_type) , intent(in)  :: fldList
+    character(len=*)             , intent(in)  :: stdname_in
+    integer                      , intent(out) :: fldindex_out
+
+    ! local variables
+    integer :: n
+    character(len=*), parameter :: subname='(med_fldList_GetFldInfo_index)'
+    ! ----------------------------------------------
+
+    fldindex_out = 0
```

Comment:
I don't think so.  The caller might be happy to not find the field, it will then do something else.  If we do add a check and abort, then we should add another flag to the interface to disable the check.  I'm happy to update it if we decide what's needed.